mono/metadata/w32file-unix.c

```diff
@@ -2107,6 +2107,21 @@ static mode_t convert_perms(guint32 sharemode)
 }
 #endif
 
+static gboolean already_shared(gboolean file_alread_shared, ino_t inode)
+{
+#if PLATFORM_MACOSX
+	/* On macOS and FAT32 partitions, we will sometimes get this inode value
+	 * for more than one file. It means the file is empty.  When this happens,
+	 * the hash table of file shares becomes corrupt, since more then one file
+	 * has the same inode. Instead, let's assume it is always fine to share
+	 * empty files. (Unity case 950616).
```

Comment:
Can you mention the define used in the internal header?

Reply:
I've fixed this, squashed, and done a force push.